pallets/summary/src/lib.rs

```diff
@@ -880,29 +879,16 @@ pub mod pallet {
         pub fn convert_data_to_eth_compatible_encoding(
             root_data: &RootData<T::AccountId>,
         ) -> Result<String, DispatchError> {
-            let eth_description =
-                EthAbiHelper::generate_ethereum_description_for_signature_request(
-                    &T::AccountToBytesConvert::into_bytes(
-                        root_data
-                            .added_by
-                            .as_ref()
-                            .ok_or(Error::<T>::CurrentSlotValidatorNotFound)?,
-                    ),
-                    &EthTransactionType::PublishRoot(PublishRootData::new(
-                        *root_data.root_hash.as_fixed_bytes(),
-                    )),
-                    match root_data.tx_id {
-                        None => EMPTY_ROOT_TRANSACTION_ID,
-                        _ => *root_data
-                            .tx_id
-                            .as_ref()
-                            .expect("Non-Empty roots have a reserved TransactionId"),
-                    },
-                )
-                .map_err(|_| Error::<T>::InvalidRoot)?;
+            let maybe_root_data_tx_id = match root_data.tx_id {
+                None => EMPTY_ROOT_TRANSACTION_ID,
+                _ => *root_data
+                    .tx_id
+                    .as_ref()
+                    .expect("Non-Empty roots have a reserved TransactionId"),
+            };
 
             Ok(hex::encode(EthAbiHelper::generate_eth_abi_encoding_for_params_only(
```

Comment:
I think this name is not accurate anymore. All we want is to create an encoded payload, an encoded series of bytes that represent some data relative to the transaction. It may be some of the arguments, it may be all, they may even be in a different order, so this is not really an encoding of the params of the Tx. This name does not help in understanding what the code does.

Reply:
And on that note, if all we are doing here is encoding directly the data, and we have removed the description (because the rest of the information in there, namely the `inputs`, is not necessary) then we have a simpler task. We can create the bytes array here, push the necessary data into it and encode it. It is all at the same level of abstraction and because of that does not necessarily need to call into another function.

Currently, all we are returning here is the TxId and the Root Hash. We will want to add the Expiry when that work is done, but if these three things are passed from the outside, there is no need to call another function to push them into the array before encoding.